source-dropbox/oauth2.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	"golang.org/x/oauth2"
+)
+
+var scopes = []string{"files.metadata.read", "files.content.read"}
+
+func oauth2Spec() *pf.OAuth2 {
+	AUTH_URL_TEMPLATE_FORMAT_STR := "https://www.dropbox.com/oauth2/authorize?token_access_type=offline&client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&response_type=code&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&state={{{ state }}}&scope={{#urlencode}}%s{{/urlencode}}"
+
+	spec := pf.OAuth2{
+		Provider:                  "dropbox",
+		AuthUrlTemplate:           fmt.Sprintf(AUTH_URL_TEMPLATE_FORMAT_STR, strings.Join(scopes, " ")),
+		AccessTokenUrlTemplate:    "https://api.dropboxapi.com/oauth2/token",
+		AccessTokenHeadersJsonMap: map[string]json.RawMessage{"Content-Type": json.RawMessage(`"application/x-www-form-urlencoded"`)},
+		AccessTokenBody:           "client_id={{{ client_id }}}&client_secret={{{ client_secret }}}&code={{{ code }}}&grant_type=authorization_code&redirect_uri={{{ redirect_uri }}}",
+		AccessTokenResponseJsonMap: map[string]json.RawMessage{
+			"refresh_token": json.RawMessage(`"/refresh_token"`),
+		},
+	}
+
+	return &spec
+}
+
+type Credentials struct {
+	ClientID     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+func (c *Credentials) Validate() error {
```

Comment:
I don't think this method needs to be exported

Reply:
Wait, I got mixed up with the `Validate` in `main.go`. Yeah, I agree I don't think this needs exported either.

Reply:
`config.Validate(&` does need to be exported, but I don't think Credentials.Validate()` does. And now that I'm looking at it, the `Credentials` struct also does not need to be exported.